oximeter/impl/benches/quantile.rs

```diff
@@ -0,0 +1,53 @@
+use criterion::{criterion_group, criterion_main, BenchmarkId, Criterion};
+use oximeter_impl::Quantile;
+use rand_distr::{Distribution, Normal};
+
+/// Emulates baseline code in a Python implementation of the P²
+/// algorithm:
+/// <https://github.com/rfrenoy/psquare/blob/master/tests/test_psquare.py#L47>.
+fn normal_distribution_percentile(size: i32, p: f64) -> f64 {
+    let mu = 500.;
+    let sigma = 100.;
+    let normal = Normal::new(mu, sigma);
+    let mut observations = (0..size)
+        .map(|_| normal.unwrap().sample(&mut rand::thread_rng()))
+        .collect::<Vec<f64>>();
+    float_ord::sort(&mut observations);
+    let idx = ((f64::from(size) - 1.) * p) as usize;
+    observations[idx]
+}
+
+fn normal_distribution_quantile(size: i32, p: f64) -> f64 {
+    let mu = 500.;
+    let sigma = 100.;
+    let normal = Normal::new(mu, sigma);
+    let mut observations = (0..size)
+        .map(|_| normal.unwrap().sample(&mut rand::thread_rng()))
+        .collect::<Vec<f64>>();
+    float_ord::sort(&mut observations);
+
+    let mut q = Quantile::new(p).unwrap();
+    for o in observations.iter() {
+        q.append(*o).unwrap();
+    }
+    q.estimate().unwrap()
+}
+
+fn baseline_benchmark(c: &mut Criterion) {
+    let mut group = c.benchmark_group("Baseline");
+    let size = 1000;
+
+    for p in [0.8, 0.9].iter() {
+        group.bench_with_input(BenchmarkId::new("Normal", p), p, |b, p| {
+            b.iter(|| normal_distribution_percentile(size, *p))
+        });
+        group.bench_with_input(BenchmarkId::new("Quantile", p), p, |b, p| {
+            b.iter(|| normal_distribution_quantile(size, *p))
+        });
```

Comment:
Could you explain how to interpret these results? It appears that it compares the runtime of `Quantile::estimate()` against sorting an array of floats, and I'm having a bit of trouble understanding what we learn from that.

Reply:
Answered above. Focusing on just the quantile estimate across diff p's